crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -105,15 +119,62 @@ fn click_handler(
     }
 }
 
-fn move_camera_system(
```

Comment:
I think that having a separate code for camera movement & rest of minimap interaction is not wise. For example unit sending with right clicks suffers the same issues.

I agree that moving the camera on mouse release even when mouse was pressed outside of the minimap is not ideal. Maybe we can fix this by renaming `MinimapClickEvent` to `MinimapPressEvent` and changing `click_handler()` so that it sends the event on press. That would fix the issue with minimal effort.